src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -103,158 +107,170 @@ class TezosNodeOperator(node: TezosRPCInterface) extends LazyLogging {
     }
 
   /**
-    * Gets a given block.
-    * @param network  Which Tezos network to go against
-    * @param hash     Hash of the given block
-    * @return         Block
+    * Fetches operations for a block, without waiting for the result
+    * @param network   Which Tezos network to go against
+    * @param blockHash Hash of the block
+    * @return          The [[Future]] list of operations
     */
-  def getBlock(network: String, hash: String): Try[TezosTypes.Block] =
-    node.runGetQuery(network, s"blocks/$hash").flatMap { jsonEncodedBlock =>
-      Try(fromJson[TezosTypes.BlockMetadata](jsonEncodedBlock)).flatMap { theBlock =>
-        if(theBlock.header.level==0)
-          Try(Block(theBlock, List[OperationGroup]()))    //This is a workaround for the Tezos node returning a 404 error when asked for the operations or accounts of the genesis blog, which seems like a bug.
-        else {
-          getAllOperationsForBlock(network, hash).flatMap{ itsOperations =>
-            itsOperations.length match {
-              case 0 => Try(Block(theBlock, List[OperationGroup]()))
-              case _ => Try(Block(theBlock, itsOperations.flatten))
-            }
-          }
-        }
-      }
-    }
+  def asyncGetAllOperationsForBlock(network: String, blockHash: String): Future[List[OperationGroup]] =
+    node.runAsyncGetQuery(network, s"blocks/$blockHash/operations")
+      .map(ll => fromJson[List[List[OperationGroup]]](ll).flatten)
+
+  /**
+    * Fetches a single block from the chain, without waiting for the result
+    * @param network   Which Tezos network to go against
+    * @param hash      Hash of the block
+    * @return          the block data wrapped in a [[Future]]
+    */
+  def asyncGetBlock(network: String, hash: String): Future[TezosTypes.Block] =
```

Comment:
We definitely should get this PR out first, this will change a lot about how I structure my code for this module (not that that is a bad thing!)